core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1315,6 +1294,80 @@ private[spark] object SparkSubmitUtils {
 
 }
 
+private [deploy] object DependencyUtils {
```

Comment:
Do you mind moving this to a separate source file?

I think there's other stuff in `SparkSubmit` that should eventually be moved here, but it's ok to leave that for later.

Also, nit: no space after `private`.